pkg/validation/app.go

```diff
@@ -256,6 +256,22 @@ func (v *Validator) validateKubeConfig(ctx context.Context, cr v1alpha1.App) err
 }
 
 func (v *Validator) validateLabels(ctx context.Context, cr v1alpha1.App) error {
+	// This is for migration towards managing App CRs in the org namespace.
+	// For the transition time being we must retain backward compatibility for the
+	// App CRs in cluster namespaces.
+	isManagedInOrg := !key.InCluster(cr) && key.IsInOrgNamespace(cr)
+
+	var validationMethod func(context.Context, v1alpha1.App) error
+	if isManagedInOrg {
+		validationMethod = v.validateOrgLabels
+	} else {
+		validationMethod = v.validateClusterLabels
+	}
+
+	return validationMethod(ctx, cr)
```

Comment:
```suggestion
	if isManagedInOrg {
		return v.validateOrgLabels(ctx, cr)
	}

	return v.validateClusterLabels(ctx, cr)
```

I missed this earlier not enough ☕  :) but I think this can be simpler.

Reply:
Nice, looks more elegant 👏